### PR TITLE
reap_identity should take batch of accounts

### PR DESCRIPTION
**Pull request by dharjeezy**
closes https://github.com/paritytech/polkadot-sdk/issues/2964

Polkadot address: 12GyGD3QhT4i2JJpNzvMf96sxxBLWymz4RdGCxRH5Rj5agKW

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5386744

**Comment by joepetrowski** [comment]
I am not sure about this PR at this stage. I see the issue discussion so sorry if we've misguided you. Also curious what @ggwpez thinks.

My issue is that we've already tested this on Rococo and Westend and it is in the 1.7.0 release that will go on Kusama. That means this change would most likely apply only to Polkadot. Of course bug fixes should go to Polkadot but I'm not as keen to throw untested optimizations at it.

This is certainly better, but it's code that's only used once and then will be removed from Polkadot. The main advantage is that it saves the sender a transaction fee when using `utility.batch`. Personally, I'd pay the transaction fee for the peace of mind that it's worked on two testnets and Kusama.

File: polkadot/runtime/common/src/identity_migrator.rs [hunk 1]
```diff
@@ -112,20 +113,31 @@ pub mod pallet {
...
+
+			Self::deposit_event(Event::IdentityReaped { who_vec });
 			let post = PostDispatchInfo {
 				actual_weight: Some(<T as pallet::Config>::WeightInfo::reap_identity(
-					registrars, subs,
+					total_registrars,
+					total_subs,
 				)),
-				pays_fee: Pays::No,
+				pays_fee: Pays::Yes,
```
Comment (ggwpez) [inline polkadot/runtime/common/src/identity_migrator.rs:140]: This is not really correct, as we want to omit fee payment in the success case.  
Normally we go for something like >=90% of good values justifies a free transaction.

**Comment by ggwpez** [comment]
I supposed that it would be a trivial change (just one more for loop), but now i see that we would also need to calculate ratios again and reject empty vectors.  
But if we dont actually need it, then maybe not.

File: polkadot/runtime/common/src/identity_migrator.rs (hunk 1 above)
Comment (dharjeezy) [reply polkadot/runtime/common/src/identity_migrator.rs:140]: how do i identify good values to use as a justification of free transaction?

**Pull request closed without merging**